src/module/ovs/event.rs

```diff
@@ -0,0 +1,68 @@
+use anyhow::Result;
+use serde::{Deserialize, Serialize};
+
+use super::bpf::*;
+use crate::{
+    core::events::{bpf::BpfRawSection, *},
+    EventSection, EventSectionFactory,
+};
+
+#[derive(Default, Deserialize, Serialize, EventSection, EventSectionFactory)]
+pub(crate) struct OvsEvent {
```

Comment:
Flattening all of the sub-events into one big one makes it quite difficult to read and maintain. Even leading to repeated fields. This event is really semantically an enum, so I believe we can implement it as one using enum tagging, eg: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=e1fd08fd86e7db9004630942c4b84a14

I can give it a try and send a patch that implements this proposal. It can then be squashed into this one, or just added at the end. WDYT?

Reply:
The rust struct that best represents the OvsEvent is:

```rust

#[derive(Deserialize, Serialize)]
#[serde(tag = "action")]
enum ActionData {
   Output(u32),
   Drop,
   Recirc(u32),
...
}

#[derive(Deserialize, Serialize)]
struct Action {
   action: ActionData,
   recirc_id: u32
}

#[derive(Deserialize, Serialize)]
#[serde(tag = "event_type")]
enum OvsEventData {
    #[serde(rename = "upcall")]
    UpcallEvent(UpcallData),
    #[serde(rename = "action_execute")]
    ActionEvent(Action),
    #[serde(rename = "recv_upcall")]
    RecvUpcallEvent(RecvUpcallData),
    #[serde(rename = "flow_operation")]
    OvsOperationEvent(OperationData),
    #[serde(rename = "upcall_enqueue")]
    UpcallEnqueueEvent(UpcallEnqueueData),
    #[serde(rename = "event_type")]
    UpcallReturnEvent(UpcallReturnData),
}

```
In fact, this probably allows us to reuse most of the structs in `bpf.rs`.

Reply:
OK. Will send a follow up patch.

Reply:
I've sent #112 based on this. Do you want to merge both PRs into one?